src/providers/export/azureCustomVision.test.ts

```diff
@@ -1,26 +1,42 @@
-import axios from "axios";
-import { AzureCustomVisionProvider, IAzureCustomVisionOptions, NewOrExisting } from "./azureCustomVision";
+import shortid from "shortid";
+import _ from "lodash";
+import { AzureCustomVisionProvider, IAzureCustomVisionExportOptions, NewOrExisting } from "./azureCustomVision";
 import registerProviders from "../../registerProviders";
 import { ExportProviderFactory } from "./exportProviderFactory";
 import MockFactory from "../../common/mockFactory";
-import { IProject } from "../../models/applicationState";
+import { IProject, AssetState, IAsset, IAssetMetadata, RegionType, IRegion } from "../../models/applicationState";
 import { ExportAssetState } from "./exportProvider";
+jest.mock("./azureCustomVision/azureCustomVisionService");
+import {
+    AzureCustomVisionService, IAzureCustomVisionProject,
+    IAzureCustomVisionImage, IAzureCustomVisionTag,
+} from "./azureCustomVision/azureCustomVisionService";
+
+jest.mock("../../services/assetService");
+import { AssetService } from "../../services/assetService";
+import HtmlFileReader from "../../common/htmlFileReader";
 
 describe("Azure Custom Vision Export Provider", () => {
     let testProject: IProject = null;
 
     beforeEach(() => {
-        testProject = MockFactory.createTestProject("TestProject");
-        testProject.exportFormat = {
-            providerType: "azureCustomVision",
-            providerOptions: {},
-        };
-
-        axios.post = jest.fn(() => Promise.resolve({
-            data: {
-                id: expect.any(String),
+        jest.resetAllMocks();
+        testProject = {
+            ...MockFactory.createTestProject("TestProject"),
+            assets: {
+                "asset-1": MockFactory.createTestAsset("1", AssetState.Tagged),
+                "asset-2": MockFactory.createTestAsset("2", AssetState.Tagged),
+                "asset-3": MockFactory.createTestAsset("3", AssetState.Visited),
+                "asset-4": MockFactory.createTestAsset("4", AssetState.NotVisited),
             },
-        }));
+            exportFormat: {
+                providerType: "azureCustomVision",
```

Comment:
Nit: Instantiate provider options in MockFactory

Reply:
@tbarlow12, why is this not recommended?

Reply:
Not completely sure we should put anything in MockFactory.  This configuration seems specific to azure-custom-vision so probably it will not be reused by other tests.

Reply:
@JacopoMangiavacchi - There are many cases were we are reusing a lot of mock data especially for core components / concepts like projects, connections, assets, etc.  Agree there should likely be a rule where if we're not going to reuse it - then it may not be worth putting on the mock factory.

Reply:
It's not a huge deal either way. We've just run into quite a few scenarios where test data specific to one test gets re-used in another area. Example: I needed to test the filtering of the connections based on the provider type. So while it wasn't specific to any specific provider, I needed instantiations of connections for each provider to validate that it was filtering properly. I can see a similar use case happening here when validating export logic